### ISSUE TITLE
Fail to Deploy device-virtual with mqtt-bus and got error "Failed to subscribe Metadata system event"

### ISSUE THREAD

**Issue by cherrycl** [edgexfoundry/device-sdk-go#1293]
# 🐞 Bug Report

### Affected Services [**REQUIRED**]

 The issue is located in: device-service

### Is this a regression?

 Yes

### Description and Minimal Reproduction [**REQUIRED**]
1. Deploy EdgeX with attached compose file

## 🔥 Exception or Error
<pre><code>
level=ERROR ts=2023-01-30T10:00:50.167998044Z app=device-virtual source=main.go:126 msg="Failed to subscribe Metadata system event: An error occured while performing a 'Subscribe' operation: invalid Topic; multi-level wildcard must be last level"

</code></pre>

## 🌍  Your Environment
**Deployment Environment:** x86/ubuntu

**EdgeX Version [**REQUIRED**]:** main

**Anything else relevant?**

[docker-compose-mqtt-bus.yml.zip](https://github.com/edgexfoundry/device-sdk-go/files/10534601/docker-compose-mqtt-bus.yml.zip)

**Comment by cloudxxx8** [comment]
it looks like this kind of topic subscription doesn't work for MQTT
https://github.com/edgexfoundry/device-sdk-go/blob/881f10918eb4e60f546b60d89ac5bea6bad2c109/example/cmd/device-simple/res/configuration.toml#L77

**Comment by lenny-intel** [comment]
Correct, when switching to MQTT, the topic must be changed to `SystemEventTopic = "edgex/system-events/core-metadata/device/#+/device-simple/#" `

We have an issue on go-mod-messaging to make this the standard across all implementation. But for now we can adjust this in compose builder.

**Comment by cloudxxx8** [comment]
@lenny-intel should it be just `"edgex/system-events/core-metadata/device/+/device-simple/#"`?
https://www.hivemq.com/blog/mqtt-essentials-part-5-mqtt-topics-best-practices/

What does `#+` mean?

refer to https://github.com/edgexfoundry/go-mod-messaging/issues/166

**Comment by lenny-intel** [comment]
> What does `#+` mean?

@cloudxxx8 , that was a typo. Should just be `+` ;-)
